src/BugsnagUnity/Delivery.cs

```diff
@@ -175,14 +204,24 @@ IEnumerator PushToServer(IPayload payload)
                 if (code == 200 || code == 202)
                 {
                     // success!
-                    _payloadManager.PayloadSendSuccess(payload);
+                    _payloadManager.RemovePayload(payload);
                     StartDeliveringCachedPayloads();
                 }
-                else if (req.isNetworkError || code == 0 || code == 408 || code == 429 || code >= 500)
+                else if ( code == 0 || code == 408 || code == 429 || code >= 500)
                 {
                     // sending failed with no network or retryable error, cache payload to disk
                     _payloadManager.SendPayloadFailed(payload);
                 }
+                else if (req.isNetworkError)
```

Comment:
This does just the same as the final `else` - do you really want a specific `else if` for it?

Reply:
I'm a professional programmer, i swear!